File: 45-jump-game-ii/jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        left = 0
        right = 0
        steps = 0

        while right < len(nums) - 1:
            maxIdx = 0
            for i in range(left, right + 1):
                maxIdx = max(maxIdx, i + nums[i])
            left = right + 1
            right = maxIdx
            steps += 1
        
        return steps

File: 45-jump-game-ii/jump-game-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
    let left = 0;
    let right = 0;
    let steps = 0;
    while (right < nums.length - 1) {
        let maxIdx = right;
        for (let i = 0; i < right + 1; i++) {
            maxIdx = Math.max(maxIdx, i + nums[i]);
        };
        left = right + 1;
        right = maxIdx;
        steps++;
    }
    return steps
};